src/snowflake/snowpark/ml/transformer.py

```diff
@@ -0,0 +1,360 @@
+        else:
+            for fc, c in zip(self._states_table_cols, self._input_cols):
+                if not sparse:
+                    onehotarray = array_construct(
+                        df[f"{fc}_count"], df[f"{fc}_encoder"], lit(1)
+                    )
+                else:
+                    onehotarray = df[f"{fc}_encoder"]
+                df = df.with_column(f"{c}_onehot", onehotarray)
+                originalC.append(f"{c}_onehot")
+            df = df.select([col(c) for c in originalC])
+        return df
+
+
+class OrdinalEncoder(Transformer):
+    def __init__(
+        self,
+        input_cols: Optional[List[str]] = None,
+        output_cols: Optional[List[str]] = None,
+        category: Union[str, List[List[str]]] = "auto",
+    ):
+        super().__init__(input_cols, output_cols)
+        self.category = category
+        self.merge = False
+        self._states_table_cols = (
+            [f"states_{c}" for c in self._input_cols] if self._input_cols else None
+        )
+        self._encoder_table_cols = (
+            [f"states_{c}_encoder" for c in self._input_cols]
+            if self._input_cols
+            else None
+        )
+        self._states_table_name = random_name_for_temp_object(TempObjectType.TABLE)
+        self._category_table_name = {}
+
+    @property
+    def input_cols(self) -> List[str]:
+        return self._input_cols
+
+    @input_cols.setter
+    def input_cols(self, value: List[str]):
+        self._input_cols = value
+        self._states_table_cols = (
+            [f"states_{c}" for c in self._input_cols] if self._input_cols else None
+        )
+        self._encoder_table_cols = (
+            [f"states_{c}_encoder" for c in self._input_cols]
+            if self._input_cols
+            else None
+        )
+
+    @property
+    def output_cols(self) -> List[str]:
+        return self._output_cols
+
+    @output_cols.setter
+    def output_cols(self, value: List[str]):
+        self._output_cols = value
+
+    def fit(self, df: DataFrame) -> "OrdinalEncoder":
+        if self._output_cols is None or len(self._output_cols) == 1:
+            self.merge = True
+        elif len(self._input_cols) == len(self._output_cols):
+            self.merge = False
+        else:
+            raise ValueError("number of output colum and input column does not match")
+        encoder_fitter(self, df, "ordinal")
+        return self
+
+    def transform(
+        self,
+        df: DataFrame,
+    ) -> DataFrame:
+        originalC = df.columns
+        encoder_structure = array_construct()
+        for fc, ec, (c, tabname) in zip(
+            self._states_table_cols,
+            self._encoder_table_cols,
+            self._category_table_name.items(),
+        ):
+            cattable = df._session.table(tabname)
+            df = cattable.join(df, df[c] == cattable[fc])
+            encoder_structure = array_append(encoder_structure, col(ec))
```

Comment:
encoder_structure is only used when `self.merge` is true, right? So you should only do it when `self.merge` is true.